### DISCUSSION TITLE
Automating RTMP Streaming

### DISCUSSION THREAD

**Discussion by bruco1987** [gopro/OpenGoPro#428] [Q&A]
Hello everyone,

I am an app developer and am looking to integrate RTMP streaming functionality with a GoPro in my app without using the GoPro Quick app. I discovered that GoPro has released the Open GoPro APIs and am curious if anyone has had experience using these APIs to set up the RTMP server URL and start streaming directly from a custom app.

Additionally, I've read that there's a possibility to achieve some control over the GoPro settings using QR codes through GoPro Labs. Has anyone tried this method to set up RTMP streaming?

Here's what I am aiming to accomplish:
1. Set the RTMP server URL and streaming key directly from my app.
2. Start and stop the RTMP streaming without having to open the GoPro Quick app.

Any help or guidance would be greatly appreciated. If you have code examples or relevant documentation, that would be fantastic. Thank you very much in advance for any assistance you can provide!

Warm regards,
Primo

**Comment by martincarapia** [comment]
So, I'm trying to do something similar where I set cameras to an RTMP server to be used for streaming. I've got to the point where I can configure the cameras to an access point and configure it to live stream, but I can't start live streaming. The docs [here](https://gopro.github.io/OpenGoPro/ble/features/live_streaming.html) point that you need a GoPro 12 or higher to ping for Livestream Ready Status. Why can't Gopros >9 respond with Ready Status?

 I'm confused that you can set up live streaming from your phone using the GoPro Quik but not from the SDK. 

I find it even more weird that there was a livestreaming demo before (PR #318) the GoPro Hero 12 release of September 15, 2023. 

Why did adding Hero 12 support kill live streaming for other models? 

Essentially my question is, do I need to buy a GoPro 12 in order to complete my project or is there a work around for older models.
